w2w/w2w.py

```diff
@@ -411,84 +432,84 @@ def wrf_remove_urban(
-                    dis <= disflat.sortby(disflat).isel(gridpoints=NPIX_NLC), drop=True
-                )
-                m = stats.mode(aux.values.flatten(), nan_policy='omit')[0]
-                newluse[i, j] = int(m)
+    luf_2D = luf.values.reshape(luf.shape[0], -1)
+    if orig_num_land_cat > 20:  # USING MODIS_LAKE
+        data_coord['luf_natland'] = list(
+            (luf_2D[12, :] == 0) & (luf_2D[16, :] == 0) & (luf_2D[20, :] == 0)
+        )
+        data_coord['luf_urb'] = list((luf_2D[12, :] != 0))
+    else:  # USING MODIS (NO LAKES)
+        data_coord['luf_natland'] = list((luf_2D[12, :] == 0) & (luf_2D[16, :] == 0))
+        data_coord['luf_urb'] = list((luf_2D[12, :] != 0))
+    if NPIX_AREA == None:
+        NPIX_AREA = NPIX_NLC**2
+    if NPIX_AREA > luse.size:
+        raise ValueError(
+            f'{ERROR}ERROR: The area you selected is larger than the domain size\n'
+            f'You chose an area of {NPIX_AREA} pixels and the domain is {luse.size} pixels\n'
+            f'Reduce NPIX_AREA. \n\n'
+            f'Exiting...{ENDC}'
+        )
 
-                auxg = (
-                    greenf.where(
-                        dis <= disflat.sortby(disflat).isel(gridpoints=NPIX_NLC),
-                        drop=True,
-                    )
-                    .where(aux == newluse[i, j])
-                    .mean(dim=['south_north', 'west_east'])
-                )
-                newgreenf[:, i, j] = auxg
-
-            if luf.isel(south_north=i, west_east=j, land_cat=12) > 0.0:
-                if orig_num_land_cat > 20:  # USING MODIS_LAKE
-                    dis = _calc_distance_coord(
-                        lat.where(
-                            (luf.isel(land_cat=12) == 0.0)
-                            & (luf.isel(land_cat=16) == 0.0)
-                            & (luf.isel(land_cat=20) == 0.0)
-                        ),
-                        lon.where(
-                            (luf.isel(land_cat=12) == 0.0)
-                            & (luf.isel(land_cat=16) == 0.0)
-                            & (luf.isel(land_cat=20) == 0.0)
-                        ),
-                        lat.isel(south_north=i, west_east=j),
-                        lon.isel(south_north=i, west_east=j),
-                    )
-                else:  # USING MODIS (NO LAKES)
-                    dis = _calc_distance_coord(
-                        lat.where(
-                            (luf.isel(land_cat=12) == 0.0)
-                            & (luf.isel(land_cat=16) == 0.0)
-                        ),
-                        lon.where(
-                            (luf.isel(land_cat=12) == 0.0)
-                            & (luf.isel(land_cat=16) == 0.0)
-                        ),
-                        lat.isel(south_north=i, west_east=j),
-                        lon.isel(south_north=i, west_east=j),
-                    )
+    dkd, ikd = using_kdtree(data_coord, min(luse.size, NPIX_AREA))
```

Comment:
the `dkd` is never accessed no? we can make it `_` instead maybe?

Reply:
Sounds good.